tools/lint-docstring

```diff
@@ -0,0 +1,231 @@
+
+def generate_folder_file_docstrings_dict() -> Dict[str, Dict[str, str]]:
+    """
+    Returns a dictionary containing folder name which in turn
+    is a dictionary containing files and their respective descriptions
+    """
+    total_files = extract_folder_file_structure()
+    folder_file_docstring: Dict[str, Dict[str, str]] = {}
+    for folder, files in sorted(total_files.items()):
+        folder_file_docstring[str(folder)] = {}
+        for file in files:
+            imported_file = importlib.import_module(
+                f'{folder.replace("/",".")}.{file[:-3]}'
+            )
+            extracted_docstring = str(imported_file.__doc__)
+            docstring = extracted_docstring.strip().replace("\n", " ")
+            if len(docstring) > COLUMN_WIDTHS[2]:
+                raise ValueError(
+                    f"{file} has docstring longer than {int(COLUMN_WIDTHS[2])}"
+                )
+            folder_file_docstring[str(folder)][file] = docstring
+    return folder_file_docstring
+
+
+def extract_folder_file_structure() -> Dict[str, List[str]]:
+    """
+    Returns dictionary containing folders and respective python files within them
+    """
+    folders_and_files = {}
+    for path_to_folder in glob.glob(f"{ZULIPTERMINAL}/**/", recursive=True):
+        complete_directory_path = Path(path_to_folder)
+        relative_directory_path = complete_directory_path.relative_to(ROOT_DIRECTORY)
+        if str(relative_directory_path) not in DOCSTRINGS_FOR_NO_FILE_FOLDERS:
+            files_in_directory = [
+                file.name
+                for file in complete_directory_path.glob("*.py")
+                if file.name != "__init__.py"
+            ]
+            folders_and_files[str(relative_directory_path)] = files_in_directory
+    return folders_and_files
+
+
+def extract_docstrings_from_file_overview() -> Dict[str, Dict[str, str]]:
+    """
+    Reads the developer-file-overview.md document and creates
+    a dictionary containing folder name which in turn is a dictionary
+    containing files and their respective descriptons
+    """
+    with open(DEVELOPER_DOC, "r") as file:
+        doc_data = file.readlines()
+
+    existing_doc_dict: Dict[str, Dict[str, str]] = {}
+
+    # table is present from line 6 to (length - 1)
+    for doc_row in doc_data[TABLE_OFFSET - 1 :]:
+        _, folder, filename, docstring, _ = doc_row.split("|")
+        folder, filename = folder.strip(), filename.strip()
+        if folder:
+            folder_name = folder
+            existing_doc_dict[str(folder_name)] = {}
+        if docstring.strip():
+            if filename:
+                existing_doc_dict[folder_name][filename] = docstring.strip()
+            else:
+                existing_doc_dict[folder_name]["no_file_present"] = docstring.strip()
+    return existing_doc_dict
+
+
+def pad_column_cells(
+    column_text: Tuple[str, str, str],
+    columnwise_widths: Tuple[int, int, int] = COLUMN_WIDTHS,
+) -> Tuple[str, ...]:
```

Comment:
This function is unnecessary. Python f-strings have this built in.

```pycon
>>> folder = "zulipterminal"
>>> file = "api_types.py"
>>> docstring = "Preliminary Zulip API types defined in python, to allow type checking"
>>> print(f"| {folder:23}| {file:20}| {docstring:103}|\n")
| zulipterminal          | api_types.py        | Preliminary Zulip API types defined in python, to allow type checking                                  |
```

(The widths can be computed expressions too. [Documentation](https://docs.python.org/3/reference/lexical_analysis.html#f-strings).)

Reply:
Aah, I remember using this a while back but it didn't cross my mind. Let's use this.

Reply:
Thank you for this suggestion @andersk 💯